sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -453,13 +453,160 @@ abstract class GpuRoundBase(child: Expression, scale: Expression) extends GpuBin
   override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType)
 
   override def doColumnar(value: GpuColumnVector, scale: GpuScalar): ColumnVector = {
+
     val lhsValue = value.getBase
+
+    // Fixes up integral values rounded by a scale exceeding/reaching the max digits of data
+    // type. Under this circumstance, cuDF may produce different results to Spark.
+    //
+    // In this method, we handle round overflow, aligning the inconsistent results to Spark.
+    //
+    // For scales exceeding max digits, we can simply return zero values.
+    //
+    // For scales equaling to max digits, we need to perform round. Fortunately, round up
+    // will NOT occur on the max digits of numeric types except LongType. Therefore, we only
+    // need to handle round down for most of types, through returning zero values.
+    def fixUpOverflowInts(zeroFn: () => Scalar): ColumnVector = {
```

Comment:
I would prefer defining these functions out of this `doColumnar` to making them as the internal functions.

Reply:
Done.